### DISCUSSION TITLE
Jotai in Library Development

### DISCUSSION THREAD

**Discussion by Etesam913** [pmndrs/jotai#1705] [General] [answered]
I am creating a npm library that uses jotai and one problem that I encountered was with state sharing. When having two instances of a component the two instances share the same state. 

For example,
Clicking on one button would increase the count of both buttons.

I followed the docs to try to fix this:
https://jotai.org/docs/api/core#provider

I used a Provider with a scope as shown in the docs example. However, this doesn't seem to work. The only way I got it to work was by doing what is seen in the video below: 

https://user-images.githubusercontent.com/55665282/213614026-beb9f93f-e72c-46c4-bc01-4fcb57dc720e.mp4

This is the code:
https://github.com/Etesam913/Custoplayer/blob/a404b5f663b30589ebae2532f4cce7e4a16f02d0/packages/custoplayer/src/lib/MyButton.tsx

Should the example in the docs change to match what I did in the video?

**Comment by dai-shi** [comment, marked as answer]
Your `useAtom` is not under the provider tree.

Here's the fix:
```jsx
const countAtom = atom(0);
const myScope = Symbol();

function MyButtonComponent() {
  const [count, setCount] = useAtom(countAtom, myScope);
  return (
    <StyledButton onClick={() => setCount((prev) => prev + 1)}>
      {count}
    </StyledButton>
  );
}

function MyButton() {
  return (
    <Provider scope={myScope}>
      <MyButtonComponent />
    </Provider>
  );
}
```

**Reply by Etesam913** [reply]
Thanks, that was the issue 👍🏽
